noir/noir-repo/compiler/noirc_evaluator/src/brillig/brillig_ir/entry_point.rs

```diff
@@ -284,15 +272,18 @@ impl<F: AcirField + DebugToString> BrilligContext<F, Stack> {
                     self.deallocate_single_addr(target_index);
                 }
             }
-
             self.deallocate_register(movement_register);
-
-            deflattened_array_pointer
         } else {
-            let deflattened_array_pointer = self.allocate_register();
-            self.mov_instruction(deflattened_array_pointer, flattened_array_pointer);
-            deflattened_array_pointer
+            self.codegen_mem_copy(
```

Comment:
We should be able to avoid mem copying for flat array arguments if we issue multiple calldatacopy, instead of one. I'll address that after this PR.